hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/conf/Configuration.java

```diff
@@ -3247,7 +3249,15 @@ private void handleInclude() throws XMLStreamException, IOException {
           File href = new File(confInclude);
           if (!href.isAbsolute()) {
             // Included resources are relative to the current resource
-            File baseFile = new File(name).getParentFile();
+            File baseFile;
+
+            try {
+              baseFile = new File(new URI(name));
+            } catch (IllegalArgumentException | URISyntaxException e) {
+              baseFile = new File(name);
+            }
+
+            baseFile = baseFile.getParentFile();
             href = new File(baseFile, href.getPath());
```

Comment:
I was worried here about what if baseFile = null at this point, but java.io.File is happy with that.